server/routerlicious/packages/lambdas/src/scribe/summaryWriter.ts

```diff
@@ -515,16 +515,20 @@ export class SummaryWriter implements ISummaryWriter {
             logTail;
 
         // Check the missing operations in the fullLogTail
-        if (fullLogTail.length !== (to - from - 1)) {
+        const fullLogTailSequenceNumbers = fullLogTail?.map((ms) => ms.sequenceNumber);
+        const isSameSNInterval = fullLogTailSequenceNumbers &&
```

Comment:
Maybe my eyes are too tired, but I'm having some trouble understanding the logic to compute `isSameSNInterval`. Perhaps we can add some comments to help understand this, which should also help in the long term?

The fact that we later use the opposite of it (that is, `if (!isSameSNInterval)`) also made me a bit confused. Would it be possible to simplify the logic by inverting the if condition? Not sure if that is possible - I was just brainstorming, so please ignore if that might be too complicated 😄

Reply:
Sounds good! I just added the comment to explain the logic.